Working branch

IanEisenberg opened the pull request.

IanEisenberg committed:
added functions to load results, extract particular batteries and experiments and calculate basic timing statst

IanEisenberg committed:
started making results class. Currently select_experiments isn't working correctly

IanEisenberg committed:
more work on the Results class. Changed clean_df function slightly

IanEisenberg committed:
updated and cleaned up results class. Moved some functions outside of class and added a filter method

IanEisenberg committed:
moved jspsych specific functions to jspsych_utils

vsoch commented on expanalysis/jspsych_utils.py:
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""
```
You would want jspych utils to be a module within expfactory-analysis, eg:

expanalysis/
   experiments/
       jspsych.py
   utils.py

That way the user would do something like

```
  from expanalysis.experiments.jspsych import ...   
```

IanEisenberg commented:
I was actually just looking this up. How do I make these functions modules within expfactory-analysis? I see you have things like "from expanalysis.maths import check_numeric", but I can't figure out how to import files like that.

Also, you think jspsych-utils.py should be under experiments? Not on the same level as utils.py (so expanalysis/jspsych-utils?)

vsoch commented:
> > I was actually just looking this up. How do I make these functions modules within expfactory-analysis? I see you have things like "from expanalysis.maths import check_numeric", but I can't figure out how to import files like that.

"maths" would be a file called maths.py, and check_numeric a function within it.

> > Also, you think jspsych-utils.py should be under experiments? Not on the same level as utils.py (so expanalysis/jspsych-utils?)

Yes, jspsych is a kind of experiment, so it should be a module in the experiments folder. You could imagine that since we will have games and surveys, we will also have folders for those, eg:

```
   \expanalysis
        \experiments
            jspsych.py
        \surveys
        \games
            phaser.py
```

vsoch commented on expanalysis/results.py:
```diff
@@ -0,0 +1,166 @@
+"""
+Created on Thu Mar 24 16:03:04 2016
+
+@author: ian
```
You should replace these headers (from your IDE?) with something akin to what we have for the other modules

IanEisenberg replied:
Got it!

On Monday, March 28, 2016, Vanessa Sochat notifications@github.com wrote:

> In expanalysis/results.py
> https://github.com/expfactory/expfactory-analysis/pull/6#discussion_r57655934
> :
> 
> > @@ -0,0 +1,166 @@
> > +# -_\- coding: utf-8 -_-
> > +"""
> > +Created on Thu Mar 24 16:03:04 2016
> > +
> > +@author: ian
> 
> You should replace these headers (from your IDE?) with something akin to
> what we have for the other modules
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/expfactory/expfactory-analysis/pull/6/files/3892e3b7f763661f207c7a1035e5a1b105749329#r57655934

## 

Best,
Ian

vsoch commented on expanalysis/results.py:
```diff
@@ -0,0 +1,166 @@
+        self.data = self.select_battery(battery)
+        
+    def results_to_df(self):
+        if isinstance(self.data_json,list):
```
take a look at the other modules to see how to document the functions, eg with :param variable: and a description - this is important for when we have Sphinx docs, so the function details get parsed.

IanEisenberg committed:
stated a temporary workspace. Changed clean_df to accept a dictionary of columns/rows to drop

IanEisenberg committed:
routine

vsoch commented on expanalysis/jspsych.py:
```diff
@@ -1,15 +1,14 @@
-"""
-Created on Mon Mar 28 16:07:54 2016
+'''
+analysis/results.py: part of expfactory package
```
expanalysis package :)

vsoch commented on expanalysis/results.py:
```diff
@@ -1,9 +1,8 @@
 
-@author: ian
-"""
+'''
```
This looks much better! I think what you want here is more of a validate function - before any operations are done on a result object, you would want to confirm it looks ok. The validate should check for battery, experiments, etc, then when you write any kind of function that uses a result, you can just call the validate function first thing, and it will take care of printing out errors / alerting the user about a misformed result object.

IanEisenberg committed:
added more documentation, fixed compute_contrast, added compute_regression

IanEisenberg committed:
changed analysis to expanalysis in header

IanEisenberg commented:
I added a validate function, but not sure if I'm capturing the functionality you're envisioning. All functions should just require battery, experiment and worker, so I added that validation to the results object itself (in case it somehow loads something else).

vsoch commented:
Don't worry about it, I can take a detailed look probably next weekend, and integrate into the API. Thanks for your hard work!

IanEisenberg committed:
changed imports to local imports (should be changed to module like imports in the final iteration). Starting working on general getter functions like get_average_variable and get_post_task_responses

IanEisenberg committed:
added more experiments to get_drop_rows lookup table

IanEisenberg committed:
routine

IanEisenberg committed:
allowed results object to be created from file. Added data export function

IanEisenberg committed:
Changed how time_taken was calculated by adding in instruction time. 'reduce_word' function added to deal with synonyms

IanEisenberg committed:
changed import to correct format

IanEisenberg committed:
added a template filter and check for jspsych. updated some functions to work with multiple templates, or check for the correct template

IanEisenberg committed:
updated functions

IanEisenberg committed:
further additions to account for different data types

IanEisenberg committed:
minor additions

IanEisenberg committed:
more updates

IanEisenberg committed:
update workspace

IanEisenberg committed:
updated requirements, more routine work

IanEisenberg committed:
more updates

IanEisenberg committed:
routine

IanEisenberg committed:
added processing module and refactored structure. Started adding post-processing functions

IanEisenberg committed:
worked on plotting

IanEisenberg committed:
fixed plotting

IanEisenberg committed:
updated data_check function to worker across workers

IanEisenberg committed:
updated temp wip

IanEisenberg committed:
added keep_track post

IanEisenberg committed:
routine

IanEisenberg committed:
removed unnecessary filter dictionary functionality

IanEisenberg committed:
updated util to read surveys and output value text

IanEisenberg committed:
removed response text option for now - broken

IanEisenberg committed:
merged

IanEisenberg committed:
added an analyiss template folder

IanEisenberg committed:
added DV functions

IanEisenberg committed:
more changes

IanEisenberg commented:
I have another PR that integrates the new results object into these functions and have moved them to the experiments folder to separate them from the base expanalysis.
